### DISCUSSION TITLE
How do you print a variable's address in Carbon?

### DISCUSSION THREAD

**Discussion by JushBJJ** [carbon-language/carbon-lang#1780] [General]
In C, printing a simple address would be like:
```
#include <stdio.h>

int main(void) {
    char *str1 = "This is a string literal";
    char *str2 = str1;

    printf("%p", str2);

    return 0;
}

```

In Carbon, doing similar will return an error.
```

package sample api;

fn Main() -> i32{
        let str1: String = "test";
        let str2: auto = &str1;

        Print("%p", str2);
        return 0;
}

```

Error: `COMPILATION ERROR: ./address.carbon:5: Argument to & should be an lvalue.`

**Comment by geoffromer** [comment]
[`let` declares a variable as an r-value](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/README.md#binding-patterns), which means that it doesn't have an address. To declare a variable that has an address, use `var` instead of `let`.

**Reply by JushBJJ** [reply]
I see thanks for that 👍
`Print` doesn't seem to support outputting addresses?

**Reply by geoffromer** [reply]
Yeah, that hasn't been implemented yet. I'm not sure implementing it should be a priority right now, because Explorer represents addresses in a very different way than a compiler would, so the output is likely to be confusing.
